plasmapy_nei/eigen/eigenclass.py

```diff
@@ -0,0 +1,354 @@
+"""
+Contains the class to provide access to eigenvalue data for the
+ionization and recombination rates.
+"""
+
+__all__ = ["EigenData"]
+
+import h5py
+import numpy as np
+from numpy import linalg as LA
+import pkg_resources
+import warnings
+
+try:
+    from plasmapy.particles import Particle, particle_input
+except (ImportError, ModuleNotFoundError):
+    from plasmapy.atomic import Particle, particle_input
+
+max_atomic_number = 30  # TODO: double check if this is the correct number
+
+
+def _get_equilibrium_charge_states(ioniz_rate, recomb_rate, natom):
+    """
+    Compute the equilibrium charge state distribution for the
+    temperature specified using ionization and recombination rates.
+
+    Parameters
+    ----------
+    """
```

Comment:
I agree with all of these.  I'm going to slightly address these now, and raise an issue or two so that the refactoring will be done in a follow-up pull request.  I also think this would be more readable if it were broken up into smaller functions, which may reduce the need for inline clarification comments.